manifests/server.pp

```diff
@@ -60,7 +60,7 @@
   Optional[Integer] $slowms                             = undef,
   Optional[Stdlib::Absolutepath] $keyfile               = undef,
   Optional[String[6]] $key                              = undef,
-  Optional[String] $set_parameter                       = undef,
+  Optional[Array[String]] $set_parameter                = undef,
```

Comment:
this would be a backwards-incompatible change. can you keep support for strings?
```suggestion
  Variant[Optional[String[1]],Array[String[1]]] $set_parameter = undef,
```

(this is untested)

Reply:
I've pushed an update so that now the logic is both backward-compatible with configurations that provide string. @bastelfreak if this version is fine I will squash the commits and rebase onto a fresh master.

Reply:
is there a reason for net setting `String[1]`, but `String`? I don't think that empty strings make sense here?

Reply:
Changed it to `String[1]` for both cases. Considering most of the other parameters mostly don't specify that I didn't think that's nessercery. But why not really